packages/core/admin/admin/src/content-manager/pages/EditView/components/DocumentActions.tsx

```diff
@@ -0,0 +1,236 @@
+          onClick={secondaryAction.onClick}
+          justifyContent="center"
+          variant="secondary"
+        >
+          {secondaryAction.label}
+        </Button>
+      ) : null}
+    </Flex>
+  );
+};
+
+/* -------------------------------------------------------------------------------------------------
+ * DocumentActionComponents
+ * -----------------------------------------------------------------------------------------------*/
+
+const PublishAction: DocumentActionComponent = ({ activeTab, id, model, collectionType }) => {
+  const navigate = useNavigate();
+  const { formatMessage } = useIntl();
+  const canPublish = useDocumentRBAC('PublishAction', (state) => state.canPublish);
+  const { publish } = useDocumentActions();
+  const modified = useForm('UpdateAction', ({ modified }) => modified);
+  const isSubmitting = useForm('PublishAction', ({ isSubmitting }) => isSubmitting);
+
+  return {
+    disabled: !canPublish || isSubmitting || activeTab === 'published' || modified,
+    label: formatMessage({
+      id: 'app.utils.publish',
+      defaultMessage: 'Publish',
+    }),
+    onClick: async () => {
+      if (id) {
+        const res = await publish({
+          collectionType,
+          model,
+          id,
+        });
+
+        if ('data' in res) {
+          navigate({ search: `?status=published` });
+        }
+      }
+    },
+  };
+};
+
+PublishAction.type = 'publish';
+
+const UpdateAction: DocumentActionComponent = ({ activeTab, id, model, collectionType }) => {
+  const navigate = useNavigate();
+  const { formatMessage } = useIntl();
+  const { canCreate, canUpdate } = useDocumentRBAC('PublishAction', ({ canCreate, canUpdate }) => ({
+    canCreate,
+    canUpdate,
+  }));
+  const { create, update } = useDocumentActions();
+
+  const isSubmitting = useForm('UpdateAction', ({ isSubmitting }) => isSubmitting);
+  const modified = useForm('UpdateAction', ({ modified }) => modified);
+  const setSubmitting = useForm('UpdateAction', ({ setSubmitting }) => setSubmitting);
+  const document = useForm('UpdateAction', ({ values }) => values);
+
+  return {
+    disabled:
+      Boolean((!id && !canCreate) || (id && !canUpdate)) ||
+      isSubmitting ||
+      !modified ||
+      activeTab === 'published',
+    label: formatMessage({
+      id: 'content-manager.containers.Edit.save',
+      defaultMessage: 'Save',
+    }),
+    onClick: async () => {
+      setSubmitting(true);
+      try {
+        if (id) {
+          await update(
+            {
+              collectionType,
+              model,
+              id,
+            },
+            {
+              id,
+              ...document,
+            }
+          );
+        } else {
+          const res = await create(
+            {
+              model,
+            },
+            document
+          );
+
+          if ('data' in res) {
+            /**
+             * TODO: refactor the router so we can just do `../${res.data.id}` instead of this.
```

Comment:
💯 we have the same issue with the history page's back button

Reply:
Nice, one of us could do this in the cleanup phase of v5 👍🏼